dashboard/config/linux/upstream-apparmor-kasan.config

```diff
@@ -506,6 +504,7 @@ CONFIG_CMDLINE="earlyprintk=serial net.ifnames=0 sysctl.kernel.hung_task_all_cpu
 CONFIG_MODIFY_LDT_SYSCALL=y
 # CONFIG_STRICT_SIGALTSTACK_SIZE is not set
 CONFIG_HAVE_LIVEPATCH=y
+CONFIG_X86_BUS_LOCK_DETECT=y
```

Comment:
I see there is some sysctl for runtime enabling. What's the default mode?
I think we want to enable detection of kernel split locks (and ignore userspace, maybe we can select killing user-space).
Split locks in the kernel is a pretty bad performance bug that we could detect.

Reply:
There are two parameters. One is the `split_lock_detect` sysctl.
https://kernel.googlesource.com/pub/scm/linux/kernel/git/next/linux-next/+/refs/heads/master/arch/x86/kernel/cpu/bus_lock.c#48

I don't see how setting `split_lock_mitigate` to `0` could help us. It prints a warning in either case:
https://kernel.googlesource.com/pub/scm/linux/kernel/git/next/linux-next/+/refs/heads/master/arch/x86/kernel/cpu/bus_lock.c#220

***

There's also a `split_lock_detect` cmdline option, which I guess we want to set to `warn`
https://kernel.googlesource.com/pub/scm/linux/kernel/git/next/linux-next/+/refs/heads/master/arch/x86/kernel/cpu/bus_lock.c#120

`fatal` also sends `SIGBUS` to userspace apps that cause split locks, but I think we don't want that.
https://kernel.googlesource.com/pub/scm/linux/kernel/git/next/linux-next/+/refs/heads/master/arch/x86/kernel/cpu/bus_lock.c#387

Reply:
I don't see any actual `WARN` calls in there. `pr_warn_once` at best.

Reply:
Even for kernel? Then we won't detect them, or we need additional pattern in pkg/report.

Reply:
> or we need additional pattern in pkg/report

Looks like that

Reply:
Let's at least file an issue for this, or we will never recover this knowledge.

Reply:
Filed https://github.com/google/syzkaller/issues/5228